<!--
## Introduction
### Comprehensive characterization of biological systems with multi-omics
 - Single cell community has advanced technologies to enable concurrent processing of biological systems at multiple molecular resolutions
 - The lack of prior knowledge and gold standard benchmark naturally leads to a data-driven approach
<!--
### New single cell multi omics initiatives:
 - Human Cell Atlas (HCA): assess variation in normal tissues
 - Brain initiative and Allen Brain
 - Human Tumor Atlas Network (HTAN): Single-cell, longitudinal, and clinical outcomes atlases of cancer transitions for diverse tumor types.
<!--
### What bulk multi-omics (e.g. TCGA, ENCODE) have taught us:
 - Type of omics that can answer a specific biological question
 - The value of open resources for methodological developments
 - New hypotheses
<!--
### Using hackathons to illustrate analysis standards and challenges for capturing biological information from multi-omics technologies
 - Brief overview of our three hackathon studies highlighting state of the art challenges (e.g., spatial transcriptomics, cross-study analysis, epigenetic regulation)
 - Challenges include issues with noise and experimental design, Time lag between regulatory levels not addressed and many open questions
remain (e.g methylation / gene expression), Direction of regulation not captured
 - We present our findings from hackathon case studies that helped us obtain benchmarks and define a common language for multi-omics
<!--
- **Objectives of this paper**
    - Provide guidelines on tools / data / technologies / methods and needs to model the multi-scale regulatory processes in biological systems for a computational biologist audience
<!--
- **Outline and messages**
    - Cellular and molecular regulation is fundamentally multi-scale and captured by distinct data modalities
    - Traditional hypothesis-driven multi-omics/view studies only consider one facet of these technologies, but more can be learned through a holistic approach extending into atlases
    - We present our findings from hackathon case studies that helped us obtain a broader picture and language
    Outline of the paper:

-->

## Multi-omics hackathon studies to illustrate standards and challenges

The rapid pace of technological development can be seen as only six years after single-cell sequencing was name method of the year, single-cell multimodal omics claimed the title [@https://www.nature.com/articles/s41592-019-0703-5].  Multi-omics technologies provide a unique opportunity to fully characterize biological systems at the spatial and molecular levels. While each data modality can uniquely resolve specific biological scales and biological questions, complementary data integration techniques can resolve multi-scale interactions. At the same time, multi-modal omics technological advances have coincided with the formation of tremendous new data resources and the formation of Atlas based initiatives to characterize biological systems. Despite the potential of single cell multi-omics technologies, computational techniques and benchmark strategies to integrate datasets across high-throughput measurement technologies remain an active area of research. Thus, a central question is what are the optimal analysis methods that can be carried forward from these data from which to learn novel biological processes.

The hackathon studies we selected for the Mathematical Frameworks for Integrative Analysis of Emerging Biological Data Workshop tailored independent challenges of data integration. 
The first challenge included spatial molecular profiling. While this technology is rapidly emerging, it often provides lower molecular resolution than non-spatial counterparts. Integration strategies that merge spatial and omics datasets have the promise to enhance the molecular resolution of spatially resolved profiling. To address this challenge, we designed a hackathon using spatially resolved transcriptional data from seqFISH with corresponding non-spatial single cell profiling data from the mouse visual cortex [@doi:10.1038/nn.4216, @doi:10.1038/nbt.4260]. 
The second challenge dealt with the limited availability of tissue to obtain multiple measurements in samples from identical conditions, raising the question as to whether information can be transferred from datasets between disparate sample cohorts. Therefore, we designed the second hackathon to contain two triple negative breast cancer cohorts profiled with single cell proteomics profiling from mass cytometry (CyTOF) [@doi:10.1016/j.cell.2019.03.005] and spatial in-situ proteomics from Multiplexed Ion Beam Imaging (MIBI) [@doi:10.1016/j.cell.2018.08.039]. 
Contrary to the latter challenges, the third challenge presented data at different molecular scales but on the same cells, to  investigate how genetic and epigenetic alterations to DNA further drive the transcriptional regulation that mediates intra- and inter-cellular signaling processes underlying cellular fate transitions and states. Thus, our third hackathon was designed with  scNMT-seq data to obtain concurrent DNA methylation, chromatin accessibility, and RNA expression from the same cells to delineate the regulatory networks that underlie mouse gastrulation [@doi:10.1093/humupd/dmy021]. 

Altogether, the analysis approaches that were employed to address these disparate hackathons across biological contexts provide an unique opportunity to identify technology-specific challenges and unifying themes that are essential to effectively employ multi-omics datasets into new biological knowledge. 


![Main challenges discussed during our brainstorming sessions from the hackathons.](images/Outline.png){#fig:outline width = 50%}

