## scRNA-seq + seqFISH as a case study for spatial transcriptomics

### Overview and biological question  

The first hackathon aimed to leverage the complemenatary strengths of sequencing and imaging based single-cell transcriptomic profiling with computational techniques to integrate scRNA-seq and seqFISH data in the mouse visual cortext. While single cells are considered the smallest units and building blocks of each tissue, they still require proper spatial and structural three-dimensional organization in order to assemble into a functional tissue that can exert its physiological function. In the last decade, single-cell RNA-seq (scRNA-seq) has played a key role to capture single cell gene expression profiles, allowing us to to virtually map all the different cell types and states in whole organisms. Despite this remarkable achievement, this technology is based on cellular dissociation and hence does not maintain spatial relationships between single cells. Emerging technologies can now profile the transcriptome of single cells within their original environment, thus offering the possibility to examine how gene expression is influenced by cell-to-cell interactions, and how it is organized in a spatially coherent manner. One such approach is sequential single-molecule fluorescence in situ hybridization (seqFISH [@doi:10.1038/nmeth.2892, @doi:10.1016/j.neuron.2016.10.001]), which can identify single molecules at (sub)cellular resolution with high sensitivity.

In contrast with scRNAseq, seqFISH and many other spatial transcriptomic technologies often pose significant technological challenges, resulting in a small number of profiled genes per cell (10-100s). The newer generation of seqFISH technology (called seqFISH+ [@doi:10.1038/s41586-019-1049-y]) has greatly enhanced its capacity to profile up to 10,000 genes, but this technology is more complex and costly than seqFISH. 

New computational approaches are needed to effectively integrate scRNAseq and seqFISH data analyses. This first hackathon provided seqFISH and scRNAseq data corresponding to the mouse visual cortex ([@doi:10.1038/nbt.4260], [@doi:10.1038/nn.4216]) and our participants were challenged to accurately identify cell types. The scRNA-seq data included transcriptional profiles at a high molecular resolution whereas the seqFISH data provided spatial characterization at a lower molecular resolution. Two key computational challenges were identified to enable high-resolution spatial molecular resolution. Firstly, we explored a number of strategies to identify the most likely cell types in the seqFISH dataset based on information obtained from the scRNAseq dataset. Secondly, we sought the opposite direction to transfer spatial information obtained from the seqFISH dataset to that of the scRNAseq dataset. Cell type labels were derived from scRNA-seq analysis [@doi:10.1038/nn.4216] and previous seqFISH/scRNA-seq integration [@doi:10.1038/nbt.4260] were also provided as reference. **Could we have a more detailed description of the data chracteristics here please (number of cells, genes per data set, any filtering applied in the hackathon, is applicable**


![](images/seqFish_mockup.png){#fig:spatial width = 60%}

Caption Figure: **Overview of seqFISH and scRNA-seq integration analysis**. 
**A** Assessment of cell type prediction using different data normalizations and classifiers. Normalization strategies included none (raw), counts per million (cpm), ComBat batch correction applied to cpm (cpm_combat), scRNAseq and seqFISH scaled using the first eigenvalue (cpm_eigen), latent variables retained for both datasets after applying Partial Least Squares regression to cpm_eigen normalized data (cpm_pls). Classifiers approaches included a supervised multinomial classifier with elastic net penalty (enet), a semi-supervised multinomial classifier with elastic net penalty (ssenet) and Support Vector Machine (SVM, supervised). Each classifier was trained using the scRNAseq data and the known (provided) cell type labels, then predicted the cell type labels in the seqFISH data; for the SVM we used the predictions from the original study (Challenge 1). The Gower distance between each method-normalization pair was computed and depicted on a multidimensional scaling plot. The first dimension (x-axis) separates methods that normalize the scRNAseq and seqFISH data together (dashed) and separately (solid), showing that normalization had a stronger impact on cell type predictions than the classification method used.
**B** SVM classification models with different C parameters were trained with different number of genes in scRNAseq data using Recursive Feature Elimination (RFE) to evaluate the minimal number of genes required for data integration. The results show that a smaller gene list than what the original study proposed was sufficeint to identify cell types in both data types (Challenge 1).
**C** LIGER was applied to combine spatial and single cell transcriptomic datasets. From the separate and integrative analyses, plots of identified and known clusters were generated and metrics of integration performance were compared, showing some loss of information as a result of the integration (Challenge 1) **Josh I need an update of that figure, could you change tasic to scRNA-seq in the caption?**.
**D** Construction of a spatial network from cells' positions using Voronoi tessellation, where cell types were inferred from SVM trained on scRNA-seq data. Left: A neighbors aggregation method computes aggregation statistics on the seqFISH gene expression data for each node and its first order neighbors to address Challenge 2. Right: Identification of spatially coherent areas that can contain one or several cell types and can be used to detect genes whose expression is modulated by spatial factors rather than cell type.


### Computational challenges

#### Challenge 1: overlay of scRNA-seq onto seqFISH for resolution enhancement

<!--
**[suggestion 1]**  
Sequencing and imaging based single-cell transcriptomic profiling have complementary strengths.
Whereas single-cell RNAseq generates transcriptome-wide information, it does not have spatial information.
On the other hand, seqFISH (Lubeck 2014; Shah 2016) provides single-cell resolution spatial information, but typically profiles the expression level of only 100-300 genes.
Although the newer generation of seqFISH technology (called seqFISH+) has greatly enhanced its capacity which can now be used to profile 10,000 genes (Eng 2019), the technology is significantly more complex and costly.
As such, it is desirable to develop computational approaches to effectively integrate scRNAseq and seqFISH data analyses.
In this hackathon, the participants were provided with seqFISH (Zhu 2018) and scRNAseq (Tasic 2016) data corresponding to the mouse visual cortex and challenged to accurately identify cell-types by integrating both datasets.
Cell type labels, derived from scRNAseq analysis (Tasic 2016) and previous seqFISH/scRNAseq integration (Zhu 2018) were also provided as reference.
-->

<!-- A variety of computational approaches were applied to achieve this goal, including: supervised classification with support vector machines (Coullomb, Xu), supervised and semi-supervised (self-training) elastic net classifiers (Singh), and unsupervised matrix factorization methods (Sodicoff) (Figure {@fig:spatial}).
While the methodologies are different, a number of themes recur, such as the importance of gene selection and batch effect correction.
As expected, the ability to identify refined cell-type structure relies on the selection of cell-type specific marker genes in seqFISH data, suggesting a potential benefit of using single-cell RNAseq data to guide seqFISH experimental design.
Batch effect is another important factor affecting the accuracy of data integration.
While a number of batch effect correction methods have been developed (COMBAT, Seurat, Scanorama, etc), it remains challenging to distinguish technical from biological variations if the biological samples do not match exactly.
-->

The mouse visual cortex consists of multiple complex cell types, however, the number of profiled genes in the seqFISH dataset was limited to 125. In addition, these genes were not prioritized based on their ability to discriminate between cell types. Assigning the correct cell identity presents an important challenge.
In contrast, the scRNAseq dataset is transcriptome wide, and includes the 125 aforementioned genes.

This challenge proposed to use all genes to identify the cell type labels for each cell in the scRNA-seq data with high certainty. Next, we leveraged that information to build a classifier based on (a subset of) the 125 common genes only that was subsequently applied to the seqFISH dataset to assign cell types to each cell. 

During the hackathon, participants aimed to test various machine learning or data integration models. However, preliminary analyses highlighted that normalization strategies had a significant impact on the final results (Figure {@fig:spatial}A). In addition, although unique molecular identifier (UMI) based scRNA-seq and seqFISH can both be considered as count data, we observed dataset specific biases that could be attributed to either platform (imaging vs. sequencing batch effects) or sample specific sources of variation. In this hackathon, we opted to apply a quantile normalization approach which forces a similar expression distribution for each shared gene.

Two classifier approaches were considered: supervised and semi-supervised generalized linear model regularized with  elastic net penalty (enet and ssenet) and support vector machines (SVM) that are supervised. The ssenet approach builds a model iteratively: it combines both datasets and initially only retains the highest confidence labels, then gradually adds more cell type labels until all cells are classified (Figure {@fig:spatial}A). This type of self-training approach might be promising to generalize to other datasets. To further improve the SVM model, several combinations of kernels and optimal hyperparameters were assessed using a combination of randomized and zoomed search. In addition, different flavors of gene selection using recursive feature elimination were considered to identify the optimal or minimal number of genes needed to correctly classify the majority of the cells (Figure {@fig:spatial}A). Finally, different classification accuracy metrics were considered to alleviate the major class imbalance in the dataset, since more than 90% of cells were excitatory or inhibitory neurons, using for example balanced classification error rate. Another approach, LIGER, based on integrative non-negative matrix factorization (NMF) was applied to integrate both datasets in a subspace based on shared factors, where cell type labels could be transferred using a nearest neighbor approach (Figure {@fig:spatial}D).

<!-- Mention limitation in the biological relevance of the genes seelcted in RFE? , but the lack of goad standard hinders our ability to evaluate the relevance of such genes -->

**could we have some elements of conclusion here?**

#### Challenge 2: Identifying spatial expression patterns at the tissue level through integration of gene expression and spatial cellular coordinates 

Most tools that were originally developed for scRNA-seq data can be adapted for spatial transcriptomic datasets (Section @ref{sec:common}), however, methods to extract sources of variation that originate from spatial factors are still lacking. Novel methods which can integrate the information obtained from gene expression with that of the spatial coordinates from each individual cell or transcript (for sub-cellular resolution) within a tissue of interest are needed.

To identify spatial expression patterns in the seqFISH dataset the participants first retained spatial information at the cellular level through the formation of a spatial network based on Voronoi tessellation ([@doi:10.1101/701680]). Within this spatial network the expression of each individual cell was spatially smoothed by calculating the average gene expression levels over all its neighboring cells. UMAP on the now smooth and aggregated data matrix was applied and identied cell clusters with a density based clustering approach (Figure {@fig:spatial}D). Interestingly, these results showed that the obtained clusters themselves are spatially separated and do not necessarily overlap with specific cell types, suggesting that the spatial dimension cannot be simply captured from the expression data only.

An unanswered question is whether the identified combinatorial spatial patterns can be extracted directly from a matching scRNA-seq dataset, similar to what was previously shown for individual gene expression profiles that can be mapped to their known spatial locations ([@doi:10.1038/nbt.3192], [@doi.org/10.1016/j.cell.2019.05.006]). However, this still constitutes both a technological and analytical challenge that will require careful benchmarking in the near future (Section @ref{sec:benchmarking}).

<!-- integrating different datasets and extrapolating observed trends is challenging and will require more work and careful benchmarking in the future. -->

<!--
**[suggestion 1]**  
How could one identify spatial patterns in the seqFISH data? This broad question can be divided into a number of specific tasks, such as detecting genes whose expression is spatially coherent, cell types whose spatial distribution is confined to distinct regions, recurrent multi-cell-type interaction clusters, etc, and the spatial scale may vary from subcellular all the way to tissue-wide organizations.
While there have been abundant studies in the geo-spatial analysis domain, computational tools targeting specific spatial transcriptomic questions are still lacking.
In this hackathon, one group tackled this challenge by aggregating gene expression data from neighboring cells followed by spatial clustering (Coullomb).
Much more work in the future is need to further explore such information.
-->

<!--
**[suggestion 2]**  
Most analyses that were originally developed for scRNAseq data can be immediately applied to spatial transcriptomic datasets, however methods to extract sources of variation that originate from spatial factors are still sparse.
To incorporate spatial information the cells from the seqFISH dataset were first connected through a spatial network based on Voronoi tessellation and then the expression of each individual cell was spatially smoothed by calculating the average gene expression levels over all the neighboring cells.
This smoothened and aggregated data matrix was subsequently used to create a 2 dimensional UMAP from which clusters were identified through a density based clustering approach.
The obtained cluster labels can then be mapped back to the original spatial locations  for further visual inspection and analysis.
-->

<!--
**[suggestion 2 detailed]**  
Most analyses that were originally developed for scRNAseq data can be adapted for spatial transcriptomic datasets, however methods to extract sources of variation that originate from spatial factors are still sparse. The latter requires the development of novel methods which can integrate the information obtained from gene expression with that of the spatial coordinates from each individual cell or transcript (for sub-cellular resolution) within a tissue of interest.
For this hackaton we aimed to incorporate the spatial information by connecting the cells from the seqFISH dataset through a spatial network based on Voronoi tessellation. Then, for each node, its RNA count data was aggregated with its first order neighbors' count data. The mean and standard deviation were computed for each gene in the gathered data in order to capture the global tendency as well as the variability in the area arround each node.
Thus, each node has `nb_genes x nb_statistics (here 2)` variables.
These "aggregation statistics" can be visualized on a 2D UMAP projection.
These data were clustered in UMAP reduced spaces of dimensionality between 2 and 9, higher dimensions allowing to define more fine-grained clusters.
The clustering was performed with HDBSCAN, a noise-aware density-based algorithm that can define arbitrary-shaped clusters.
These clusters can then be visualized on the 2D UMAP projection and on the 2D spatial map of seqFISH data.
The clusters are spatially coherent, some of them contain several cell types, and a given cell type is not necessarily limited to one specific cluster.
During the exploratory phase consisting in varying the number of dimensions and the minium cluster size, a specific spot area was found clustered for several parameters combinations, suggesting it wasn't an artifact of the choice of parameters.
"Differential expression" analysis was performed between this spot and the other areas, although we don't look at differences in gene data but in aggregation metrics.
This area seems to correspond to a "regeneration hub", but this analysis has to be considered carefully and further analyzes is required to confirm this hypothesis.
This "neighbors  aggregation" method has been extended to aggregate RNA counts (or other node attributes) to higher orders of neighbors in order to define aggregation metrics on wider areas, which could be useful for analyzes of bigger tissues.
One interesting extension would be to substract phenotypes contributions to RNA counts for each cell before performing the neighbors aggregation analysis in order to highlight genes that are modulated by spatial factors.
But if we want to retrieve the mean expression of a cluster for cells belonging to it, we should first check cluster's convexity and be sure that no other cluster lies within it.
-->
