## Spatial proteomics as a case for cross-study and cross-platform analysis

### Overview and biological question

Contrary to the first hackathon, which included seqFISH and and scRNA-seq data of samples from the same biological conditions, our second hackathon used single-cell targeted proteomics data of primary breast cancer tissue from different cohorts of patients to study the tumor-immune microenvironment in primary breast cancer. Analyzing tumor samples from different individuals involves considerable challenges, including cross-study and cross-platform integrative analysis with a low number of overlapping features (Section @ref{sec:common}). Single-cell proteomics data were generated on different antibody-based targeted proteomics technological platforms and in different laboratories. Mass cytometry (CyTOF) measured 73 proteins in two panels (immune and tumor) in 194 tissue samples from 143 subjects, of which 6 patients had triple-negative negative breast cancer [@doi:10.1016/j.cell.2019.03.005], while the second dataset applied Multiplexed Ion Beam Imaging (MIBI) to quantify spatial in-situ expression of 36 proteins in 41 triple-negative breast cancer patients [@doi:10.1016/j.cell.2018.08.039] (Figure {@fig:proteomics}AB). **what about the third study, can you describe in 1 sentence here**

<!-- Whilst this is a formidable data integration challenge, it reflects the bioinformatics analysis of clinical teams who wish to compare and investigate data collected on different measurement platforms and cohorts of cancer patients. -->
This formidable data integration task included three challenges. The first challenge was to assess whether analytical methods could integrate partially-overlapping proteomic data from different patients with similar phenotypes. In addition, we sought to determine whether measurements from one technology (MIBI spatial location and expression of proteins) could be used to predict information in the second technology (spatial expression patterns of proteins measured on mass-tag). The second challenge aimed to determine if integrated analyses of single-cell spatial technologies could capture unique information about immune cell populations in breast cancer beyond cell type composition. 
<!-- Could information about the spatial location of immune cell populations in breast cancer be discovered in integrated analyses of these datasets. -->
The third challenge addressed the critical issue of a lack of overlap between patients to assess whether heterogeneous phenotype information could be used to integrate patient omics data with a low number of shared features.

![](images/scProt_mockup.png){#fig:proteomics width = 50%}

Caption figure: **Overview of spatial proteomics cross-study and cross-platform integration analysis.** 
**A** Overview of single-cell targeted proteins hackathon challenge.
**B** Challenge 1: Partial to no overlap between protein features across studies.
**C** Challenge 2: Spatial analysis with Moran's index computed on Gabriel graph shown in boxplot according to tumor/immune status showing a significant difference between groups (Red asterisks indicate significance of an ANOVA of each group with all others with p-value from an overall ANOVA across the three groups reported).
<!--(using both dummy variables and protein expression measurements) differs significantly between groups. Boxplot of Moran’s index values on tumor/immune status with examples from Keren et al. [@doi:10.1016/j.cell.2018.08.039] corresponding to each tumor category. Red asterisks indicate significance of an ANOVA of each group with all others, and the p-value from an overall ANOVA across the three groups is reported. -->

### Computational challenges

#### Challenge 1: Low overlap between protein features across studies

There were only 20 proteins assayed in both studies (Figure {@fig:proteomics}A-B **numbers dont match, specify which study**), which precluded integration of features at the gene set or pathway level and required the use of surrogate measures for cross-study association. The majority of included proteins were cell type markers or therapeutic biomarkers, providing the opportunity to perform cross-study integration of cell type proportions in tumor tissue samples.

Several semi-supervised and supervised algorithms were applied to transfer cell labels and cell compositions from one dataset to the second. Random forest was considered to capture the hierarchical structure of cell lineage and perform feature transfer learning of cell type labels, using an adaptation of the prediction strength approach [@doi:10.1198/106186005X59243]. To assess model robustness: first, a model was trained on the labeled dataset, then used to predict labels in the unlabeled dataset; next, a second model was trained based on the second dataset with the newly predicted labels; finally, the ability of the second model to recover the correct original labels when making predictions on the labeled dataset was assessed. Mapping cells from CyTOF to imaging with spatial information was handled by solving an entropic regularization optimal transport problem [4, 11] **proper DOI ref please, in google doc**, using the cosine distance of the common proteins between the two datasets as transport cost. The constructed optimal transport plan can be considered as the likelihood of cells from one modality being mapped to cells from the other modality, which allows the prediction of protein expression measured only in CyTOF on imaging data. Cluster analysis of the resulting imputed expression matrix revealed that sub tumor cell type could be identified, which was not previously detectable in the original matrix.

This challenge raised other topical issues. The scales of protein expression limited the ability to integrate cell compositions through correlation analysis, as some protein markers were expected to be present on a range of cell types (e.g. CD45 on immune cells), while others were more specialized and represented only in a subset of those cells (e.g. CD4 on T-cell subtypes). Other limitations associated with proteomics-based cell type composition analysis included the uncertainty of antibody specificity and lack of consistency between studies, accuracy of protein markers for cell type identifcation, as well as tissue and disease heterogeneity. Cell type assignment was also seen as a major hurdle, as it relies on manually curated annotation and is dependent on domain-specific knowledge. To date, methods for cell type assignment or differential expression cannot be easily applied to targeted proteomics. Therefore, there is an urgent need for a unifying map between cells present in different datasets and for annotation resources to provide quality metric or priors of protein cell type markers. The construction of protein expression atlases would support cell type classification, even if antibodies used and their performance varies between labs.

<!--and potentially if this could be developed from  large scale consortiums IHC of proteins (Human Protein Atlas [5, 6]), although the antibodies used and their performances might vary between labs. -->

<!--Standards/QC/Normalization -->


#### Challenge 2: spatial protein expression analysis

CyToF mass spectrometry data provided protein expression and counts/composition of cells in the breast tumor-immune environment, while the MIBI-TOF data provided spatial information that quantified cell attributes (e.g. shape, size, spatial coordinates) in addition to expression levels. These datasets allowed us to examine protein expression within the tumor microenvironment to predict cell-cell interactions.

Spatial information can be encoded as a set of XY coordinates (cell centroid), a line (e.g. tumor-immune boundary), or a polygon that can define complex shapes such as a cell or a community of cells. Spatial protein expression can be summarized by descriptive statistics, such as the autocorrelation protein expression within a neighborhood of polygons. These statistics can be computed by techniques developed in geographical information science or ecology to assess whether a spatially measured variable has a random, dispersed, or clustered pattern [7]. **ref DOI**.

<!--The neighborhood of polygons can be defined with a Euclidean distance or sphere, by a number of bounded cells or other measures, many of which were developed in geographical information science or ecology and assess if a spatially measured variable has a random, dispersed or clustered pattern [7].
-->

We used a variety of approaches to investigate whether protein expression data could be used to predict the spatial properties of tissue samples. A K-nearest neighbor graph was used to build spatial response variables and a random forest model was trained on protein expression data to predict spatial features. Topic modeling was trained on protein expression and cell compositions in the CyToF data to predict cell co-locations in a fraction of MIBI-ToF considered as test data (10%),or vice versa. Among the five topics identified, the first topic was predominant in most of the immune cells from CyTOF data, while the other four were dominant in all other cells. We examined the prognostic performance of different higher level spatial metrics using Moran’s Index with a sphere distance, cell type localization using nearest neighbor correlation, or cell type interaction composition with Ripley's L-function. Cox models with fused lasso penalty and random forest survival models were then fitted based on clinical features such as patient age, tumor stage, grade, size, and cell type composition. We found that the spatial metrics were predictive of prognosis, which was particularly notable in triple-negative breast cancer, where the clinical features are considered poorly prognostic. Further investigation into spatial metrics using a graph-based neighborhood measure (Gabriel graph, based on Delaunay triangulation) found that Moran’s I differed significantly between the three prognostic tumor scores described [@doi:10.1016/j.cell.2018.08.039](Figure {@fig:proteomics}C). Thus, this challenge emphasizes the need to develop new spatial measures specific for single-cell spatial proteomics data. **Note: we will need to link back to the vignettes**

<!--
Note from KA: I had to cut back a lot, but understood very little of what was described originally.
Kris Sankaran examined the extent to which expression data could be used to predict spatial properties of tissue samples.
To build predictors, cells were first clustered (K = 20) on the basis of protein expression.
Sample-level expression summaries were defined as the proportions of cells belonging to each cluster.
To build the spatial response variables, a K-nearest neighbor graph was obtained from cell centroids (K = 5).
For each cell, the average distance to its 5 nearest neighbors was computed, reflecting its local density.
Further, the entropy of the cluster memberships across nearest neighbors was found, reflecting local heterogeneity.
To summarize samples’ cell ecosystems, cell-level statistics were averaged across each sample’s cells.
A random forest model trained from expression to spatial predictors achieved an average cross validation RMSEs of (tk ) for neighborhood size and entropy, respectively, relative to baselines of () obtainable by predicting the mean.
Dr. Pratheepa Jeganathan applied topic modelling and defined five topic trained on protein expression and cell compositions in the CyToF data were sufficient to predict cell co-locations, in 10% MIBI-ToF Test data.
Pratheepa Jeganathan (Stanford) applied a Bayesian modelling approach based on latent Dirichlet allocation (Blei, Latent dirichlet allocation Journal of machine Learning research 3.Jan (2003): 993-1022).
Topic modeling was used to identify the dominated topics and assign
spatial location of MIBI-TOF cells to the CyTOF data or vice-versa, based on the topic distribution in each cell (**Ref topic modelling?**).
Among the five topics identified, the first topic was dominated in most of the immune cells from CyTOF data and the other four dominated in all other cells.
Cells from MIBI-TOF were depicted in five clusters (link to vignette) and were consistently based on the observed and predicted marker expression, but these clusters were not identified with only observed marker expressions.
[ further details from Pratheepa available in pdf file in debrief folder]
Yingxin Lin (Sydney) examined the prognostic performance of different higher level spatial metrics.
She measured protein autocorrelation using Moran’s Index (I)  with a sphere distance, cell type localisation using nearest neighbour correlation, or cell type interaction composition, Ripley's L-function.
High-dimensional Cox models with fused lasso penalty and random forest survival models were fitted utilising different features, including clinical features such as tumour stage, tumour grade, age and tumour size, as well as features like cell type composition.
Evaluating by the c-index via cross-validation, the spatial metrics are found to be predictive, especially in triple negative breast cancer where clinical features such as grade are poorly prognostic.
Lauren Hsu (Harvard) also considered Moran’s I but used a graph-based neighborhood measure (Gabriel graph, based on Delaunay triangulation) instead of a sphere euclidean distance, and found that Moran’s I differed significantly between the three prognostic tumor scores described by Keren, et al. [@doi:10.1016/j.cell.2018.08.039](Figure {@fig:proteomics}C).
Need for development of spatial measure  - different in dimensions of RNA v proteins
-->

#### Challenge 3: Fourth corner Integration of data at the level of phenotype

Cross-study integration also raises the challenge of non-overlapping biological samples with similar phenotypes. We aimed to identify biomarkers predictive of phenotype from the different data types and explore the agreement between biomarkers selected across multiple datasets. Such biomarkers should enable the extension of biological knowledge that is not available by single omics data **please reword to be more specific - ie extention to what**. To solve this third challenge, we used phenotypic data (such as the cell attributes) to link the two datasets (Figure {@fig:common}D).

We first integrated clinical phenotype measures such as grade, stage, and overall **How can you say it was successful? on which basis?**. However, the inherent differences between datasets obtained from distinct approaches and the limited number of overlapping proteins posed an extreme challenge for integration. Only 13 proteins were shared between datasets **numbers dont match, specify which study**. Borrowing from ecology and the French school of ordination, this problem can be described as a case of the fourth corner problem (or RLQ, Figure {@fig:common}D). Briefly, given two omics datasets with available phenotypic data and non-overlapping features and samples, multiplying the two phenotypic factors should derive a bridging matrix that links the features of two datasets. This requires the two phenotypic matrices to be multipliable, i.e. describing the same phenotypical factors. While not attempted in this hackathon, we hypothesize that the fourth corner RLQ could be solved by using matrix decomposition methods [@doi:10.1007/BF02427859; doi:10.1111/ecog.02302]. **was it too hard or lack of time?**

<!--
The participants were successful at data integration using patient phenotype measures such as grade, stage and overall survival.
Breast cancer is highly heterogeneous, and multiple breast cancer molecular subtypes have been described [8, 9].
Both MIBI and Jackson data used different approaches to cell type annotation and had 13 proteins in common.
-->
<!-- Yingxin. MIBI, Jackson data.  13 proteins in common and both had used different approaches to cell type annotation, optimal transport.  KM curve Fig 1e?  
benchmarking datasets - Jackson. Assessment of biological relevance/success of methods -->

<!-- Borrowing from ecology and the French school of ordination, Chen Meng (Munich) described this problem as a case of the fourth corner problem (or RLQ).
Briefly, given two omics data where both rows (features) and columns (samples) are non overlapping, and phenotypical data available for each omics data, multiplying the two phenotypical factors will derive a bridging matrix that links the features of two omics data.
We should note that the two phenotypical matrices need to be multipliable, i.e. the phenotypical data should describe the same phenotypical factors over the samples in the corresponding dataset.
The Chessel fourth corner RLQ is a matrix decomposition method to solve the problem [@doi:10.1007/BF02427859; doi:10.1111/ecog.02302].
It decomposes the bridging matrix (phenotypical matrix) into components, each of which often represents a specific phenotypical pattern in the data.
The loading matrix of each of the omics data indicate how a feature is correlated with phenotypical factors.
-->
