## Spatial proteomics and cross-study analysis

### Overview and biological question
##  Spatial proteomics and cross-study analysis

Overview and Biological Question
The single-cell targeted proteomics hackathon investigated the tumor-immune microenvironment of primary breast cancer tissue. The challenge was the joint cross-study, cross-platform integrative analysis of single cell proteomics data that had low feature overlap, were generated on different antibody-based targeted proteomics technological platforms and in different laboratories [1, 2].  Mass cytometry (CyTOF)  from Wagner et al. (2019) measured 73 proteins in two panels (immune, tumor) in 194 tissue samples from 143 subjects, of which 6 patients had triple-negative negative breast cancer.  The second dataset applied Multiplexed Ion Beam Imaging (MIBI)  to quantify spatial in-situ expression of 36 proteins in 41 triple-negative breast cancer patients  (Keren et al. 2018).  (Figure 3a) 

Whilst this is a formidable data integration challenge, it reflects the bioinformatics analysis of clinical teams who wish to compare and investigate data collected on different cohorts of cancer patients. The first questions simply asked if partially-overlapping proteomic data collected on different patients with similar phenotypes could be integrated, and asked participants to test if measurements in one technology could be transferred and used to predict information in the second. The MIBI data provided spatial location and expression of proteins, therefore the challenge asked if the spatial expression patterns of proteins measured on mass-tag could be predicted.  Secondly it raised questions unique to spatial ‘omics technologies.  Does spatial capture unique information, beyond cell compositions. Could information about the spatial location of immune cell populations in breast cancer be discovered in integrated analyses of these datasets. Finally the datasets had no overlap in patients, so how could heterogeneous phenotype information be used to integrate patient ‘omics data with low feature and no tumor biological sample overlap. 

### Computational challenges

#### Challenge 1: address the lack of overlap between proteins across studies
#### Challenge 2: spatial protein expression analysis
