## Spatial proteomics and cross-study analysis

### Overview and biological question

The single-cell targeted proteomics hackathon investigated the tumor-immune microenvironment of primary breast cancer tissue. The challenge was the joint cross-study, cross-platform integrative analysis of single cell proteomics data that had low feature overlap, were generated on different antibody-based targeted proteomics technological platforms and in different laboratories [1, 2].  Mass cytometry (CyTOF)  from Wagner et al. (2019) measured 73 proteins in two panels (immune, tumor) in 194 tissue samples from 143 subjects, of which 6 patients had triple-negative negative breast cancer.  The second dataset applied Multiplexed Ion Beam Imaging (MIBI)  to quantify spatial in-situ expression of 36 proteins in 41 triple-negative breast cancer patients  (Keren et al. 2018).  (**Figure 3A**) 

Whilst this is a formidable data integration challenge, it reflects the bioinformatics analysis of clinical teams who wish to compare and investigate data collected on different cohorts of cancer patients. The first questions simply asked if partially-overlapping proteomic data collected on different patients with similar phenotypes could be integrated, and asked participants to test if measurements in one technology could be transferred and used to predict information in the second. The MIBI data provided spatial location and expression of proteins, therefore the challenge asked if the spatial expression patterns of proteins measured on mass-tag could be predicted.  Secondly it raised questions unique to spatial ‘omics technologies.  Does spatial capture unique information, beyond cell compositions. Could information about the spatial location of immune cell populations in breast cancer be discovered in integrated analyses of these datasets. Finally the datasets had no overlap in patients, so how could heterogeneous phenotype information be used to integrate patient ‘omics data with low feature and no tumor biological sample overlap. 



<img src="https://github.com/BIRSBiointegration/whitePaper/blob/master/content/images/scProt_mockup.png" alt="Figure 3" height="400"/>
**Figure 3**. **A** sc_targeted proteins hackathon challenge (**Kris to simplify or merge with B**)
**B** Challenge 1: Lack of overlap between protein features across studies  (**update with color friendly color**)
**C** Challenge 2: Spatial analysis with Moran's index: Moran’s index is a computed on Gabriel graph (using both dummy variables and protein expression measurements) and differs significantly between groups. Figure shows boxplot of Moran’s index values on tumor/immune dummy variable with examples from Keren et al. (2018) corresponding to each tumor category. Red asterisks indicate significance of an ANOVA of each group with all others, and the p-value from an overall ANOVA across the three groups is reported above.
**D** Challenge 3: Partial overlap of data (**Kris to avoid overlap with common challenge Fig 5 or remove**)


### Computational challenges

#### Challenge 1: Lack of overlap between protein features across studies
The low number of features precluded integration of features at the level of gene set or pathways. There were only 20 proteins that were assayed in both studies (**Figure 3A-B**). The majority of features were cell-type markers or biomarkers targets of breast cancer therapeutic intervention. The limited overlap in these studies necessitated the use of surrogate measures of cross- study association.   

Although the overlap in proteins was low, many proteins were cell-type markers, providing the opportunity to perform cross-study integration of cell type proportions in tumor tissue samples. Hackathon participants applied several semi-supervised and supervised algorithms to transfer cell labels and cell compositions from one dataset to the second.  To capture the hierarchical structure of cell lineage, Lauren Hsu (Harvard) applied a simple random forest (RF) approach to perform feature transfer learning of cell type labels. An adaptation of the prediction strength approach described in Tibshirani (2012 https://doi.org/10.1198/106186005X59243) demonstrated model robustness: first, a model was trained on the labeled dataset and used to predict labels in the unlabeled dataset; next, a second model was trained based on the second dataset with the newly predicted labels; finally, she assessed the ability of the second model to recover the correct original labels when making predictions on the labeled dataset. 

Yingxin Lin (Univ of Sydney) mapped the cells from CyTOF to imaging with spatial information by solving an entropic regularization optimal transport problem [4, 11], utilising the cosine distance of the common proteins between the two datasets as transport cost. The constructed optimal transport plan can be considered as likelihood of cells from one modality mapped to cells from the other modality, which allows the prediction of protein expression measured only in CyTOF on imaging data. By clustering on the imputed expression matrix, she was able to identify a sub tumour cell type that is not revealed in the original matrix. 

However the scales of protein expression was a possible limiting factors When integrating cell compositions using the correlation expression of protein markers,  some cell markers were expected on a range of cell types (e.g., CD45), whereas others are more specialized and represent a subset of those cells  (e.g., CD4).   Others challenges associated with cell compositions analysis of proteomics analysis included uncertainty over antibody specificity and consistency between studies, the sensitivity and specificity of protein markers for cell types, tissue and disease heterogeneity

The assignment of cell type relied on manually curated protein annotation, and was dependent on domain-specific knowledge; for example that CD4 is expressed by T-cells.   To date, methods for cell type assignment, classification or extraction of differentially expressed proteins cannot easily be applied to targeted proteomics.  Participants expressed a need for a unifying map between cells present in different datasets, and for annotation resources to provide quality metric or priors of protein cell type markers. There is a need for protein expression atlases to support cell type classification and potentially if this could be developed from  large scale consortiums IHC of proteins (Human Protein Atlas [5, 6]), although the antibodies used and their performances might vary between labs.  

Standards/QC/Normalization


#### Challenge 2: spatial protein expression analysis

Whereas the CyToF mass spectrometry provided protein expression and counts/composition of cells in breast tumor-immune environment, the MIBI-TOF data provided spatial information that quantified cell attributes (shape/size/spatial coordinates) in addition to expression levels, thus providing the opportunity to examine protein expression, cell microenvironment, and predict cell-cell interactions and the cellular community ecosystem. 

Spatial information can be encoded as a set of XY coordinates (cell centroid), a line (eg tumor-immune boundary) or a polygon, which is a closed plane defined by a number of lines. A polygon can define complex shapes such as a cell or a community of cells. Spatial protein expression can be summarized using spatial descriptive statistics, such as the autocorrelation of the expression of a protein within  a neighborhood of polygons.  The neighborhood of polygons can be defined with a Euclidean distance or sphere, by a number of bounded cells or other measures, many of which were developed in geographical information science or ecology and assess if a spatially measured variable has a random, dispersed or clustered pattern [7].  

Kris Sankaran examined the extent to which expression data could be used to predict spatial properties of tissue samples. To build predictors, cells were first clustered (K = 20) on the basis of protein expression. Sample-level expression summaries were defined as the proportions of cells belonging to each cluster. To build the spatial response variables, a K-nearest neighbor graph was obtained from cell centroids (K = 5). For each cell, the average distance to its 5 nearest neighbors was computed, reflecting its local density. Further, the entropy of the cluster memberships across nearest neighbors was found, reflecting local heterogeneity. To summarize samples’ cell ecosystems, cell-level statistics were averaged across each sample’s cells. A random forest model trained from expression to spatial predictors achieved an average cross validation RMSEs of (tk ) for neighborhood size and entropy, respectively, relative to baselines of () obtainable by predicting the mean.

Dr. Pratheepa Jeganathan applied topic modelling and defined five topic trained on protein expression and cell compositions in the CyToF data were sufficient to predict cell co-locations, in 10% MIBI-ToF Test data.   Pratheepa Jeganathan (Stanford) applied a Bayesian modelling approach based on latent Dirichlet allocation (Blei, Latent dirichlet allocation Journal of machine Learning research 3.Jan (2003): 993-1022). Topic modeling was used to identify the dominated topics and assign 
spatial location of MIBI-TOF cells to the CyTOF data or vice-versa, based on the topic distribution in each cell (**Ref topic modelling?**).  Among the five topics identified, the first topic was dominated in most of the immune cells from CyTOF data and the other four dominated in all other cells. Cells from MIBI-TOF were depicted in five clusters (link to vignette) and were consistently based on the observed and predicted marker expression, but these clusters were not identified with only observed marker expressions. [ further details from Pratheepa available in pdf file in debrief folder] 
 
Yingxin Lin (Sydney) examined the prognostic performance of different higher level spatial metrics.  She measured protein autocorrelation using Moran’s Index (I)  with a sphere distance, cell type localisation using nearest neighbour correlation, or cell type interaction composition, Ripley's L-function. High-dimensional Cox models with fused lasso penalty and random forest survival models were fitted utilising different features, including clinical features such as tumour stage, tumour grade, age and tumour size, as well as features like cell type composition. Evaluating by the c-index via cross-validation, the spatial metrics are found to be predictive, especially in triple negative breast cancer where clinical features such as grade are poorly prognostic.  


Lauren Hsu (Harvard) also considered Moran’s I but used a graph-based neighborhood measure (Gabriel graph, based on Delaunay triangulation) instead of a sphere euclidean distance, and found that Moran’s I differed significantly between the three prognostic tumor scores described by Keren, et al. (**Figure 3C**).

Need for development of spatial measure  - different in dimensions of RNA v proteins

#### Challenge 3: Fourth corner Integration of data (at the level of phenotype)

Another question often faced in cross-study integration is integration of biological samples that are non-overlapping but have similar phenotypes. The aim is to identify biomarkers from the different omics data to predict the same phenotype, and, more importantly, to explore how the markers selected from multiple datasets are in agreement with or distinct from each other. The integration of the markers from each dataset should enable to extend biological knowledge that is not available by single omics data. To solve the challenge, phenotypical data (such as the cell attributes) are the critical factors that should be used to link the two datasets (**Figure 3D**).

The participants were successful at data integration using patient phenotype measures such as grade, stage and overall survival. Breast cancer is highly heterogeneous, and multiple breast cancer molecular subtypes have been described [8, 9].  Both MIBI and Jackson data used different approaches to cell type annotation and had 13 proteins in common.  

<!-- Yingxin. MIBI, Jackson data.  13 proteins in common and both had used different approaches to cell type annotation, optimal transport.  KM curve Fig 1e?  
benchmarking datasets - Jackson. Assessment of biological relevance/success of methods -->

Borrowing from ecology and french school of ordination, Chen Meng (Munich) described this problem as a case of the fourth corner problem (or RLQ). Briefly, given two omics data where both rows (features) and columns (samples) are non overlapping, and phenotypical data available for each omics data, multiplying the two phenotypical factors will derive a bridging matrix that links the features of two omics data. We should note that the two phenotypical matrices need to be multipliable, i.e. the phenotypical data should describe the same phenotypical factors over the samples in the corresponding dataset. The Chessel fourth corner RLQ is a matrix decomposition method to solve the problem  (https://doi.org/10.1007/BF02427859 , https://doi.org/10.1111/ecog.02302). It decomposes the bridging matrix (phenotypical matrix) into components, each of which often represents a specific phenotypical pattern in the data. The loading matrix of each of the omics data indicate how a feature is correlated with phenotypical factors.

### Summary
**this could be included in other section**

In contrast to traditional fine resolution of mapping individual genes or features between studies, the proteomics challenge investigated hierarchical structure among the ‘omics, cell and phenotype layers and applied a number of measures of higher order concordance to integrate cross study.  Different questions asked from different angles based on the dataset:Integration, clinical, spatial

Extraction “real” and “abstract feature” space were applied in data integrations. Abstract lower dimensional representations of spatial coordinates successfully captured higher level cellular structure and were more prognostics that individual feature information, suggesting that new measures of the tumor or cell ecosystems of interacting cells are needed because these interactions are fundamental to disease progression.   These efforts will require standardized vocabulary, benchmarked methods, and common abstracted variables that can be compared between studies. .  Vital to these will be defining a new cell and cell community annotations, .. **DUNCAN**

The field will need to define vocabulary and relationships between different scales, when integrating high definition fine-resolution feature level, or subcellular molecular data with global, coarse or lower-resolution bulk data, however integrating of spatial information and of data across scale is not unique to biology.  Similar spatial and scales issues have been addressed in environmental ecology, weather science and geographical information system analysis. 

Future Limitation of channels - Single cell mass spec v spatial antibody  


