## Commonalities between analytical multi-omics approaches for hackathons  [Figure 5 + Table 1]

We summarize the common main challenges faces across all hackathons, and the common approaches adopted to analyse multi-omics single cell data.

### Summary of hackathon study-specific methods
 - **Table** describes method, foundation in the context of previous bulk and single cell literature, and technology dependence
    - Attempts to tweak existing methods and challenges associated in hackathons
    - List methods that are either technology dependent (e.g. spatial) vs universal and how to choose them


### Dependence on pre-processing method and/or variable selection
- These steps are key and affect downstream analyses
  - Normalization / data transformation (seqFish), pre-processing, gene summaries (scNMT-seq) to variable selection (seqFish)
  - reproducibility difficult / no consensus. e.g. Alexis selected 19 genes whereas Zhu original paper based on 47 genes (difference in methods / processed data)
- Hackathon data pre-processed enable better comparisons across methods
- No consensus reached as those are emerging data with no ground truth nor established biological results

### Approaches for partial overlap of information (cells / features) and how to predict (cell type, dataset) using another data set
- Overlap in each study
  - seqFish: same features but not cells; scProt: same proteins, not cells but similar patients; scNMT-seq: same cells but not features
  - How it was solved (Table)
- Anchoring information across datasets or studies is needed (Figure)
- Incorporation of existing biological knowledge
  - ‘From discovery to detection’ (Meuleman + debrief), time is ripe to include more knowledge in our data driven approaches
- Challenge: Partial cell overlap (but no features matching) and No overlap were not addressed

### Managing differences in scale and size for datasets that do not match cells or features
- Hackathons datasets did not match cells or features.
  - scNMTQ-seq: MOFA limitation when # features vary (and size of datasets).
  - seqFish: greedy approach to select the best gene subset (Alexis, size); consider batch effect removal method (Amrit, scale)

- Consensus on projection based methods, even if pre-processing was applied (Table)
- Additional weighting is needed (e.g. Arora, Abadi).

![Figure caption:
A) scSpatial: required overlap of features (genes), but cells do not overlap. Cell type prediction for seqFISH data was performed based on scRNA-seq (known) [credit: Amrit Singh / Kris S]
B) scProteomics: a small number of features overlap (proteins) but patients and cells do not overlap. Data imputation (?), spatial co-localization or cell type prediction was performed   [credit: Lauren Hsu and Pratheepan Jenagan / Kris S]
C) scNMT-seq: cells are matching across assays but features do not overlap. Data integration was performed [credit: Al Abadi / Kris S]
D) Nor cells nor features are matching  [credit: Chen Meng / Kris S]](images/summary_fig_person.png){@fig:summary-person}


Table: Different methods were used in the hackathon. * indicates the method was not applied on the hackathon data. For some common challenges, ‘bulk’ indicates the method was originally developed for bulk omics, 'sc’ indicates the method was specifically developed specifically for single cell data {#tbl:example-id}

<style type="text/css">
.tg  {border-collapse:collapse;border-spacing:0;}
.tg td{border-color:black;border-style:solid;border-width:1px;font-family:Arial, sans-serif;font-size:14px;
  overflow:hidden;padding:10px 5px;word-break:normal;}
.tg th{border-color:black;border-style:solid;border-width:1px;font-family:Arial, sans-serif;font-size:14px;
  font-weight:normal;overflow:hidden;padding:10px 5px;word-break:normal;}
.tg .tg-c8qn{background-color:#ffffff;color:#333333;font-weight:bold;text-align:left;vertical-align:top}
.tg .tg-7euo{background-color:#ffffff;color:#333333;text-align:left;vertical-align:top}
</style>
<table class="tg">
<thead>
  <tr>
    <th class="tg-c8qn"><span style="font-weight:700;font-style:normal;text-decoration:none">Common challenges</span></th>
    <th class="tg-7euo"><span style="font-weight:400;font-style:normal;text-decoration:none">Tasks</span></th>
    <th class="tg-c8qn"><span style="font-weight:700;font-style:normal;text-decoration:none">seqFISH case study</span></th>
    <th class="tg-c8qn"><span style="font-weight:700;font-style:normal;text-decoration:none">Sc targeted proteomics</span></th>
    <th class="tg-c8qn"><span style="font-weight:700;font-style:normal;text-decoration:none">scNMT-seq</span></th>
  </tr>
</thead>
<tbody>
  <tr>
    <td class="tg-7euo"></td>
    <td class="tg-7euo"><span style="font-weight:700;font-style:normal;text-decoration:none">Normalisation, data transformation, pre-processing </span></td>
    <td class="tg-7euo"><span style="font-weight:400;font-style:normal;text-decoration:none">Data distribution checks (Coullomb, Singh)</span><br><span style="font-weight:400;font-style:normal;text-decoration:none">HVG (Xu)</span></td>
    <td class="tg-7euo"><span style="font-weight:400;font-style:normal;text-decoration:none">VSN (Meng)</span><br><span style="font-weight:400;font-style:normal;text-decoration:none">Arcsinh transformation  (Jeganathan).</span><br><span style="font-weight:400;font-style:normal;text-decoration:none">Inverse transformation (Jenagan)</span><br><span style="font-weight:400;font-style:normal;text-decoration:none">Selection of patients (Jenagan)</span></td>
    <td class="tg-7euo"><span style="font-weight:400;font-style:normal;text-decoration:none">Summaries of DNA measurements (data provided in hackathon)</span><br></td>
  </tr>
  <tr>
    <td class="tg-c8qn" rowspan="4"><span style="font-weight:700;font-style:normal;text-decoration:none">Overlap</span></td>
    <td class="tg-c8qn"><span style="font-weight:700;font-style:normal;text-decoration:none">Partial feature overlap </span><span style="font-weight:400;font-style:normal;text-decoration:none">(cells not matching)</span></td>
    <td class="tg-7euo"></td>
    <td class="tg-c8qn"><span style="font-weight:700;font-style:normal;text-decoration:none">Imputation</span><span style="font-weight:400;font-style:normal;text-decoration:none">:</span><br><span style="font-weight:400;font-style:normal;text-decoration:none">Direct inversion with latent variables (Sankaran)</span><br><span style="font-weight:400;font-style:normal;text-decoration:none">Optimal transport to predict protein expression (Lin)</span><br><span style="font-weight:400;font-style:normal;text-decoration:none">KNN averaging (Jenathan*)</span><br><span style="font-weight:700;font-style:normal;text-decoration:none">No imputation:</span><br><span style="font-weight:400;font-style:normal;text-decoration:none">Biological Network Interaction (Foster*)</span></td>
    <td class="tg-7euo"><span style="font-weight:400;font-style:normal;text-decoration:none"> </span><br></td>
  </tr>
  <tr>
    <td class="tg-c8qn"><span style="font-weight:700;font-style:normal;text-decoration:none">Partial cell overlap </span><span style="font-weight:400;font-style:normal;text-decoration:none">(no features matching)</span></td>
    <td class="tg-7euo"></td>
    <td class="tg-7euo"><span style="font-weight:400;font-style:normal;text-decoration:none">MBPCA (Meng*)</span></td>
    <td class="tg-7euo"></td>
  </tr>
  <tr>
    <td class="tg-c8qn"><span style="font-weight:700;font-style:normal;text-decoration:none">No cell overlap </span><span style="font-weight:400;font-style:normal;text-decoration:none"> (partial or complete feature overlap)</span></td>
    <td class="tg-7euo"><span style="font-weight:400;font-style:normal;text-decoration:none">Averaging nearest neighbors in latent space to impute unmeasured expression values (Coullomb?)</span></td>
    <td class="tg-7euo"><span style="font-weight:400;font-style:normal;text-decoration:none">Topic modelling to predict cell spatial co-location or spatial expression (Jenathan, partial feature overlap)</span><br><span style="font-weight:400;font-style:normal;text-decoration:none">Transfer cell type label with Random Forest (Hsu, complete feature overlap?)</span></td>
    <td class="tg-7euo"><span style="font-weight:400;font-style:normal;text-decoration:none">LIGER - NMF (Welch, complete feature overlap)</span></td>
  </tr>
  <tr>
    <td class="tg-c8qn"><span style="font-weight:700;font-style:normal;text-decoration:none">No overlap </span></td>
    <td class="tg-7euo"></td>
    <td class="tg-7euo"><span style="font-weight:400;font-style:normal;text-decoration:none">RLQ (Chen*)</span></td>
    <td class="tg-7euo"></td>
  </tr>
  <tr>
    <td class="tg-c8qn" rowspan="2"><span style="font-weight:700;font-style:normal;text-decoration:none">Generic approaches </span></td>
    <td class="tg-c8qn"><span style="font-weight:700;font-style:normal;text-decoration:none">Classification &amp; feature selection</span></td>
    <td class="tg-7euo"><span style="font-weight:400;font-style:normal;text-decoration:none">Backward selection with SVM (Coullomb)</span><br><span style="font-weight:400;font-style:normal;text-decoration:none">self training ENet (Singh)</span><br><span style="font-weight:400;font-style:normal;text-decoration:none">Balanced error rate (Coullomb, Singh)</span><br><span style="font-weight:400;font-style:normal;text-decoration:none">Recursive Feature Elimination (Xu)</span><br><span style="font-weight:400;font-style:normal;text-decoration:none">(all bulk)</span></td>
    <td class="tg-7euo"></td>
    <td class="tg-7euo"><span style="font-weight:400;font-style:normal;text-decoration:none">Supervised clustering MOSAIC (Arora) (bulk)</span><br><span style="font-weight:400;font-style:normal;text-decoration:none">Lasso in regression-type models (bulk)</span><br></td>
  </tr>
  <tr>
    <td class="tg-c8qn"><span style="font-weight:700;font-style:normal;text-decoration:none">Cell type prediction</span></td>
    <td class="tg-7euo"><span style="font-weight:400;font-style:normal;text-decoration:none">Projections and clustering (Sodicoff*)</span><br><span style="font-weight:400;font-style:normal;text-decoration:none">SVM (Coullomb, Xu)</span><br><span style="font-weight:400;font-style:normal;text-decoration:none">ssEnet (Singh) </span><br><span style="font-weight:400;font-style:normal;text-decoration:none">(all bulk)</span></td>
    <td class="tg-7euo"></td>
    <td class="tg-7euo"></td>
  </tr>
  <tr>
    <td class="tg-c8qn"><span style="font-weight:700;font-style:normal;text-decoration:none">Study specific approaches</span></td>
    <td class="tg-c8qn"><span style="font-weight:700;font-style:normal;text-decoration:none">Spatial analysis</span><br><br><br><br></td>
    <td class="tg-7euo"><span style="font-weight:400;font-style:normal;text-decoration:none">HMRF</span><br><span style="font-weight:400;font-style:normal;text-decoration:none">Voronoi tesselation (Coullomb) (bulk)</span></td>
    <td class="tg-7euo"><span style="font-weight:400;font-style:normal;text-decoration:none">Spatial autocorrelation (Moran index, Hsu, Lin)</span><br><br><span style="font-weight:400;font-style:normal;text-decoration:none">Select spatial discriminative features:</span><br><span style="font-weight:400;font-style:normal;text-decoration:none">Moran Index, NN correlation, Cell type</span><br><span style="font-weight:400;font-style:normal;text-decoration:none">interaction composition, </span><br><span style="font-weight:400;font-style:normal;text-decoration:none">L function (Lin)</span><br><span style="font-weight:400;font-style:normal;text-decoration:none">(all bulk?)</span></td>
    <td class="tg-7euo"></td>
  </tr>
  <tr>
    <td class="tg-c8qn"><span style="font-weight:700;font-style:normal;text-decoration:none">Managing differences in scale</span><br></td>
    <td class="tg-7euo"><span style="font-weight:700;font-style:normal;text-decoration:none">Data integration</span></td>
    <td class="tg-7euo"><span style="font-weight:400;font-style:normal;text-decoration:none">LIGER - NMF (Sodicoff*) (sc)</span><br><span style="font-weight:400;font-style:normal;text-decoration:none">ComBat (Singh*)</span><br><span style="font-weight:400;font-style:normal;text-decoration:none">Projection methods MFA, sGCCA  (Singh*)</span><br><span style="font-weight:400;font-style:normal;text-decoration:none">UMAP / tSNE (Sodicoff*) (sc)</span></td>
    <td class="tg-7euo"><span style="font-weight:400;font-style:normal;text-decoration:none">Multi-block PCA</span><br><span style="font-weight:400;font-style:normal;text-decoration:none">Weighting matrices based on their similarities: Data wise (STATIS, MFA) (Chen*)</span><br><span style="font-weight:400;font-style:normal;text-decoration:none">(bulk)</span><br><span style="font-weight:400;font-style:normal;text-decoration:none">Scale CyTOF and MIBI-TOF to the range of CyTOF values (Jenagan)</span></td>
    <td class="tg-7euo"><span style="font-weight:400;font-style:normal;text-decoration:none">LIGER - NMF (Welch) (bulk)</span><br><span style="font-weight:400;font-style:normal;text-decoration:none">Projection to Latent Structures (Abadi) (bulk)</span><br><span style="font-weight:400;font-style:normal;text-decoration:none">MOSAIC as weighted approach (Arora) (bulk)</span></td>
  </tr>
  <tr>
    <td class="tg-c8qn"><span style="font-weight:700;font-style:normal;text-decoration:none">Inclusion of additional information </span></td>
    <td class="tg-7euo"><span style="font-weight:400;font-style:normal;text-decoration:none"> </span></td>
    <td class="tg-7euo"></td>
    <td class="tg-7euo"><span style="font-weight:400;font-style:normal;text-decoration:none">Survival prediction: Cox regression based on spatial features (Lin)</span></td>
    <td class="tg-7euo"><span style="font-weight:400;font-style:normal;text-decoration:none">Include annotated DHS Index to  anchor new/unseen data from DNase-seq, (sc)ATAC-seq, scNMT-seq, etc. for </span><span style="font-weight:400;font-style:italic;text-decoration:none">De novo</span><span style="font-weight:400;font-style:normal;text-decoration:none"> peak calling(Meuleman*) (bulk)</span></td>
  </tr>
</tbody>
</table>
