## Commonalities between analytical multi-omics approaches for hackathons {#sec:common}

Each hackathon study highlighted disparate challenges to multi-omics from different measurement technologies. However, these studies were unified by the underlying problem of data integration. We summarize the common main challenges faced across all hackathons and the common approaches participants adopted to highlight the critical computational challenges faced in multi-omics single cell data analysis.

The choice of methods mostly relied on the challenge or biological question to address: data integration was conducted using projection approaches across all studies, whilst cell prediction required machine or statistical learning methods (SVM, Enet) and spatial analysis Hidden Markov random field or Moran's Index. As the computational methodologies span technologies, so do the central challenges highlighted in each hackathon. For example the accuracy of the analysis depends critically on data pre-processing, (normalization, upstream feature selection), differences in scale across data sets and overlap (or lack thereof) features and cells (Figure {@fig:common}).

Table {@tbl:common} summarizes the main methods that were applied across all hackathons.  A large number of computational analysis methods that were applied derive from bulk RNA-seq literature, with the exception of projection methods such as tSNE, UMAP and LIGER (the latter two are based on the common techniques NMF and PCA that were further developed for single cell data). In this section, we briefly highlight three common challenges faced across all hackathons.


### Common challenge 1: Dependence on pre-processing method and/or variable selection
Pre-processing steps strongly affect downstream analyses. Our participants thoroughly assessed the effect of normalization and data transformation (e.g. spatial transcriptomics, Figure {@fig:spatial}Letter), as well as preliminary feature selection (mostly on based on highly variable genes) or feature summarization (scNMT-seq study). Ease of comparisons between analyses was facilitated by providing processed input data (@ref{sec:software}, but even such step did not avoid facing  reproducibility issues between the original published study and the new analyses. For example in the spatial transcriptomics study, Coullomb selected 19 genes (*in scRNA-seq? or seqFISH?*) whereas the original paper selected 47 genes based on the same feature selection process [@doi:10.1038/nbt.4260]. No consensus was reached across participants analyses regarding the best way to process such emerging data, as no extensive benchmark, ground truth nor established biological results are yet available (@ref{sec:benchmark}).

### Common challenge 2: Managing differences in scale and size across datasets
Different types of techniques were used to address the differences in scale or resolution across data sets. For spatial transcriptomics and proteomics, participants focused on a common set of genes (via feature selection in spatial transcriptomics) or proteins. The scNMT-seq study that included overlap between cells raised the issue of differences in data set size with a varying number of features per dataset ranging from 6673 to 18345, Figure {@fig:scNMT-seq}). Some projection-based methods, such as MOFA [@doi:https://doi.org/10.1186/s13059-020-02015-1], require a similar number of features in each data set, whilst others such as PLS / sGCCA [@doi:10.1093/biostatistics/kxu001] are not limited by such data setting and enable flexible analysis (Abadi). Difference in data scale may result in one data set contributing to either too much variation or noise during data integration. Techniques such as re-scaling (Jenagan), batch effect removal approaches, such as Combat [@doi:10.1093/biostatistics/kxj037] (Singh), or weighting specific data sets (Arora, Abadi) were considered and all offer further improvement in the analyses.


### Common challenge 3: Addressing partial overlap of information across cells or features
Degree of feature or cell overlap between datasets varied dramatically within each study. Intuitively, one requires at least one type of overlap (whether on the features, or on cells, Figure {@fig:common}) in order to integrate information across modalities. The field has made progress in developing methods to integrate data sets across the same (bulk) samples of single cells, mostly based on dimension reduction techniques. Amongst them, NMF (LIGER), Projection to Latent Structures (sGCCA [@doi:10.1093/biostatistics/kxu001]) were used for the scNMT-seq study. When there was no cell overlap (spatial studies), imputation methods were used to predict gene, protein or spatial expression values based on nearest neighbors, latent variables or optimal transport, or to predict cell types (Hsu). The most challenging study was the spatial proteomics, which raised the issue of no overlap between cells or features - the so called fourth corner that relies on phenotypes ( @ref{sec:proteomics}). We anticipate that this scenario will be avoided once technological progress and increase in data sets availability is achieved [@doi: 10.1186/s13059-020-1926-6].


### Summary
The three hackathon analyses emphasized that regardless of the common challenges faced by our participants, there is no one method fits all. Methods derived from RNA-seq are generally not sufficient for single cell data, and further developments are needed to fully suit these emerging data. It is important that the challenges highlighted in this section are fully solved to enable subsequent interpretation of multi-scale biological systems from multi-omics data. Results interpretation of these analyses, however, raise further challenges.


![
**A.** Overlap of features (genes) but not cells (e.g. spatial transcriptomics where cell type prediction for seqFISH data was performed based on scRNAseq where cell types are known.
**B.** Partial overlap of features (proteins) but no overlap of cells (e.g. spatial proteomics that required data imputation or cell type prediction).
**C.** Overlap of cells across assays, but no overlap of features (e.g. scNMT-seq where data integration was performed).
**D.** Lack of overlap between cells and features (the so-called fourth corner problem @ref{sec:proteomics}).
](images/summary_fig_person.png){#fig:common width="50%"}


Table: Different methods were used in the hackathon. * indicates the method was not applied on the hackathon data. For some common challenges, ‘bulk’ indicates the method was originally developed for bulk omics, 'sc’ indicates the method was specifically developed specifically for single cell data **table will include links to vignettes** {#tbl:common}

<style type="text/css">
  .tg {
    border-collapse: collapse;
    border-spacing: 0;
    background-color: #ffffff;
    color: #333333;
  }
  .tg td {
    border-color: black;
    border-style: solid;
    border-width: 1px;
    overflow: hidden;
    padding: 10px 5px;
    word-break: normal;
    text-align: left;
    vertical-align: top
  }
  .tg th {
    border-color: black;
    border-style: solid;
    border-width: 1px;
    overflow: hidden;
    padding: 10px 5px;
    word-break: normal;
    text-align: left;
    vertical-align: top;
    font-weight: bold;
  }
  .tg .tg-bold {
    font-weight: bold;
  }
</style>
<table class="tg">
  <thead>
    <tr>
      <th>Common challenges</th>
      <th>Tasks</th>
      <th>sc Spatial</th>
      <th>sc targeted proteomics</th>
      <th>sc NMT-seq</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td class="tg-bold">Pre-processing</td>
      <td class="tg-bold">Normalization & data transformation</td>
      <td>Data distribution checks (Coullomb, Singh)<br> High Variable Genes selection (Xu) </td>
      <td>Variance Stabilization Normalisation [@doi:10.1093/bioinformatics/18.suppl_1.s96] (Meng)<br> Arcsinh transformation (Jeganathan).<br> Inverse transformation (Jenagan)<br> Selection of patients (Jenagan) </td>
      <td>Summaries of DNA measurements (input data provided in hackathon)</td>
    </tr>
    <tr>
      <td class="tg-bold"> Managing differences in scale<br>
      </td>
      <td class="tg-bold"> Data integration </td>
      <td> LIGER [@doi:10.1101/459891] (Sodicoff) (sc)<br> ComBat (Singh)<br> Projection methods MFA, sGCCA [@doi:10.1093/biostatistics/kxu001] (Singh*) (bulk)<br> UMAP/tSNE (Sodicoff) (sc) </td>
      <td> Multi-block PCA [@doi:10.18129/B9.bioc.mogsa]<br> Weighting matrices based on their similarities: STATIS, MFA (Chen*)(bulk)<br> Scale MIBI-TOF to the range of CyTOF values (Jenagan) </td>
      <td> LIGER [@doi:10.1101/459891] (Welch) (sc)<br> Projection method sGCCA [@doi:10.1093/biostatistics/kxu001] (Abadi) (bulk)<br> Multi Omics Supervised Integrative Clustering with weights (Arora) (bulk) </td>
    </tr>
    <tr>
      <td class="tg-bold" rowspan="6">Overlap</td>
      <td><b>Cell overlap</b><br> (features not matching) </td>
      <td></td>
      <td></td>
      <td>
        <b>Dimension reduction and projection methods:</b><br> LIGER [@doi:10.1101/459891] (Welch) (sc)<br> sGCCA [@doi:10.1093/biostatistics/kxu001] (Abadi) (bulk)<br>
      </td>
    </tr>
    <tr>
      <td>
        <b>Partial feature overlap</b><br> (cells not matching)
      </td>
      <td></td>
      <td>
        <b>Imputation:</b><br> Direct inversion with latent variables (Sankaran)<br> Optimal transport to predict protein expression (Lin)<br> K Nearest Neighbor averaging (Jenathan*)</span><br>
        <br>
        <b>No imputation:</b><br> Biological Network Interaction (Foster*)
      </td>
      <td></td>
    </tr>
    <tr>
      <td><b>Partial cell overlap</b><br> (features not matching) </td>
      <td></td>
      <td>Multi block PCA [@doi:10.18129/B9.bioc.mogsa] (Meng*)</td>
      <td></td>
    </tr>
    <tr>
      <td><b>No cell overlap</b><br> (complete feature overlap) </td>
      <td>Averaging nearest neighbors in latent space to impute unmeasured expression values (Coullomb?)</td>
      <td>Transfer cell type label with Random Forest (Hsu)</td>
      <td>LIGER [@doi:10.1101/459891] (Welch)</td>
    </tr>
    <tr>
      <td><b>No cell overlap</b><br> (partial feature overlap) </td>
      <td></td>
      <td> Topic modeling to predict cell spatial co-location or spatial expression (Jenathan, partial feature overlap)<br>
      </td>
      <td></td>
    </tr>
    <tr>
      <td class="tg-bold">No overlap</td>
      <td></td>
      <td>RLQ [@doi:10.1890/13-0196.1] (Chen*)</td>
      <td></td>
    </tr>
    <tr>
      <td class="tg-bold" rowspan="4"> Generic approaches </td>
      <td class="tg-bold"> Classification &amp; feature selection </td>
      <td> Backward selection with SVM (Coullomb)<br> self training ENet (Singh)<br> Balanced error rate (Coullomb, Singh)<br> Recursive Feature Elimination (Xu)<br>
        <br> (all bulk)
      </td>
      <td></td>
      <td> Multi Omics Supervised Integrative Clustering (Arora) (bulk)<br> Lasso penalization in regression-type models (bulk) </td>
    </tr>
    <tr>
      <td class="tg-bold">Cell type prediction</td>
      <td>Projection with LIGER [@doi:10.1101/459891] (Sodicoff)<br> SVM (Coullomb, Xu)<br> ssEnet (Singh)<br> (all bulk) </td>
      <td></td>
      <td></td>
    </tr>
    <tr>
      <td class="tg-bold">Spatial analysis</td>
      <td>Hidden Markov random field<br> Voronoi tesselation (Coullomb) (bulk) </td>
      <td>Spatial autocorrelation with Moran's Index (Hsu, Lin)<br>
        <br> Selection of spatial discriminative features:<br> Moran's Index, NN correlation, Cell type, interaction composition, L function (Lin)<br>
        <br> (all bulk?)
      </td>
      <td></td>
    </tr>
    <tr>
      <td class="tg-bold">Inclusion of additional information</td>
      <td></td>
      <td>Survival prediction: Cox regression based on spatial features (Lin)</td>
      <td>Include annotated hypersensitive sites index to anchor new/unseen data from DNase-seq, (sc)ATAC-seq, scNMT-seq, for <em>de novo</em> peak calling (Meuleman*) (bulk) </td>
    </tr>
  </tbody>
</table>
