## Discussion

The analyses and reference datasets from the Mathematical Frameworks for Integrative Analysis of Emerging Biological Data Workshop demonstrated the power of hackathons to have the potential both to inform and develop new analysis methods to capture the complex, multi-scale nature of biological datasets from high-throughput data modalities. Notably, the hackathon studies of the workshop were designed specifically to span state-of-the-art multi-omics challenges to map the epigenetic, molecular, and cellular interaction across both time and sample populations. While on the one hand coupling single cell measurements can inherently simplfy the challenge of linking disparate biological scales, layering new sets of molecular measurements intrinsically increases complexity of the corresponding analyses to interpret these data. The computational needs hinge on the underlying biological question being asked as well as the data themselves. For example, different considerations must be made for cross-modal integration higlighted in the seqFISH and scNMT-seq challenges than modeling with unmatched measurements in the scProteomics challenge. Regardless, through these hackathons we identified several common analysis themes spanning algorithmic advances, interpretation, benchmarking, and software infrastructure necessary for biological interpretation. Notably, all cases require methods for dealing with data quality, data loss by summarization, timing variances both between and within omics layers, and batch effects will be necessary challenges to overcome. Conversely, the ability to infer regulatory relationships between different omics and interpret these relationship with data visualization strategies present new research opportunities.

Technologies to profile biological systems at a single cell resolution and across molecular scales are advancing at an unprecidented pace. Analytically, these advances require the computational analysis commuity to pursue research that can first enable robust, data set specific analyses that are tailored to a specific biology or measurement technology and secondly scale and adapt to these rapid advances. For example, the hackathons highlighted current technologies for spatial molecular profiling, which currently require profiling samples with multiple technologies to balance high resolution spatial and molecular profiling across data modalities. While the data are not spatially resolved, the scNMT-seq challenge highlights the potential of further inference of gene regulation through concurrent profiling of RNA, methylation, and chromatin state. Technological advances for multi-omics spatial data and epigenetics data are both rapidly advancing and becoming increasinbly commercially available through companies such as Nanostring, 10X Genomics, Akoya Biosciences, and others. The workshop keynote by Bernd Bodenmiller higlighted new research-level technological advances that further enable three dimensional spatial molecular profiling [@doi:10.1101/2020.05.24.113571] and still other technologies are exanding to allow for temporally resolved profiling [@doi:10.1038/s41592-020-0880-2]. Integration strategies that are aware of these future directions and the mathematical analysis challenges that span technologies will be most adept to advance biological interpretation from these emerging datasets. Implementing these analysis tools requires further robust software ecosystems, including as examples Bioconductor [@doi:10.1038/s41592-019-0654-x], Biopython, Scanpy [@doi:10.1186/s13059-017-1382-0], Seurat [@doi:10.1038/nbt.4096], and Giotto [@doi:10.1101/701680], in which users create these analysis approaches and anticipate stable yet adaptive data structures that are robust for these emerging technologies. The size of these emerging datasets, particularly in the context of their application to atlas projects including notably the Human Tumor Atlas Network [@doi:10.1016/j.cell.2020.03.053], Human Cell Atlas [@doi:10.7554/eLife.27041, @doi:10.1038/550451a], Allen Brain Initiative, and ENCODE <!--are there others or citations I should be using for this?--!> as key examples, comptuational efficiency and scalability of these implementations are becoming ever more critical.

In addition to new technologies, it is important to note that the analysis methods for multi-omics can in and of themselves generate new data sources to resolve the multi-scale nature of biological systems.  For example, while the workshop posed the scNMT-seq data and spatial molecular datasets as distinct challenges for data integration, integration of matched datasets between these spatial and epigenetic profiling techniques could further resolve the dependence of cell-type and cellular-interactions of regulatory networks. Further embedding prior biological knowledge as rules in the analysis approaches can provide an additional source of data from which to generate a new representation of a biological system. Curated regulatory networks from databases such as KEGG, Biocarta, GO, or MSigDB <!-- are there more resources and add citations--!> provide commonly used frameworks for this prior knowledge, but must be extended to map context-specific networks in light of emerging single cell atlases. The regulatory networks and dynamic features captured in single cell data also provide the potential for future techniques to predict future molecular and cellular states by merging single cell data with mathematical models that forecast future states using rules derived from only prior biological knowledge.
